vexim/login.php

```diff
@@ -5,35 +5,32 @@
 
   # first check if we have sufficient post variables to achieve a successful login... if not the login fails immediately
   if (!isset($_POST['crypt']) || $_POST['crypt']===''
-      || !isset($_POST['localpart']) || $_POST['localpart']===''
-      || !isset($_POST['domain'])
+      || !isset($_POST['username']) || $_POST['username']===''
   ){
     header ('Location: index.php?login=failed');
     die;
   }
 
   # construct the correct sql statement based on who the user is
-  if ($_POST['localpart'] === 'siteadmin') {
-    $query = "SELECT crypt,localpart,user_id,domain,domains.domain_id,users.admin,users.type,domains.enabled AS domainenabled FROM users,domains
+  if ($_POST['username'] === 'siteadmin') {
```

Comment:
I think the three cases can and should be merged into one.
Also, we should probably discover whether a user is site admin by checking the `type` field value, not by blindly mapping everyone whose `localpart` matches.